pynamodb/connection/base.py

```diff
@@ -923,6 +924,121 @@ def get_item(self,
+                page_size = limit
+            else:
+                page_size = read_capacity_to_consume_per_second
+
+        while True:
+            if rate_available:
+                try:
+                    data = self.scan(
+                        table_name,
+                        attributes_to_get=attributes_to_get,
+                        exclusive_start_key=last_evaluated_key,
+                        limit=page_size,
+                        conditional_operator=conditional_operator,
+                        return_consumed_capacity=TOTAL,
+                        scan_filter=scan_filter,
+                        segment=segment,
+                        total_segments=total_segments
+                    )
+                    for item in data.get(ITEMS):
+                        item_to_return = item
+                        if model_cls is not None:
+                            item_to_return = model_cls.from_raw_data(item)
+
+                        yield item_to_return
+
+                        if limit is not None:
+                            limit -= 1
+                            if not limit:
+                                return
+
+                    latest_scan_consumed_capacity = data.get(CONSUMED_CAPACITY)
+                    last_evaluated_key = data.get(LAST_EVALUATED_KEY, None)
+                    consecutive_provision_throughput_exceeded_ex = 0
+                except ScanError as e:
+                    # Only retry if provision throughput is exceeded.
+                    if isinstance(e.cause, ClientError):
+                        code = e.cause.response['Error'].get('Code')
+                        if code == "ProvisionedThroughputExceededException":
+                            consecutive_provision_throughput_exceeded_ex += 1
+                            if consecutive_provision_throughput_exceeded_ex > max_consecutive_exceptions:
+                                raise # Max threshold reached
+                        else:
+                            raise # Different exception, other than ProvisionedThroughputExceededException
+                    else:
+                        raise # Not a Client error
+
+            # No throttling, and no more scans needed. Just return
+            if not last_evaluated_key and consecutive_provision_throughput_exceeded_ex == 0:
+                return
+
+            current_time = time.time()
+
+            # elapsed_time_ms indicates the time taken in ms from the start of the
+            # throttled_scan call.
+            elapsed_time_ms = round((current_time - start_time) * 1000)
+            # elapsed_time_ms can be 0 or negative if there is a clock drift
+            if elapsed_time_ms < 1:
+                elapsed_time_ms = 1
+
+            if consecutive_provision_throughput_exceeded_ex == 0:
+                total_consumed_read_capacity += latest_scan_consumed_capacity
+                consumed_rate = total_consumed_read_capacity / elapsed_time_ms
```

Comment:
i would probably from __future__ import division at top so you don't need to worry about accidentally doing integer division